### PR TITLE
Finished comments feature

### PR DESCRIPTION
**Pull request by sarahc7**
Hi Taylor and Aditya,

I finished the comments feature. The only major addition is the new `loadComments` function in script.js. The HTML changes are mostly tag changes I made after reading documentation on layout elements. I also made a few changes to the HTML and CSS to make the site more responsive.

Here's how the comments look:
Screenshot 2020-06-30 at 8 54 18 PM

Thanks,
Sarah

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -85,7 +90,7 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
    */
   private String getParameter(HttpServletRequest request, String name, String defaultValue) {
     String value = request.getParameter(name);
-    if (value == null) {
+    if (value == null || value.equals("")) {
```
Comment (mithasgoogle) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:93]: Can you add a comment explaining why this change?

File: portfolio/src/main/webapp/style.css [hunk 2]
```diff
@@ -1,10 +1,11 @@
 a,
 button,
 input,
+label,
 p,
 span,
 textarea {
-  font: 2vw 'Titillium Web', sans-serif;
+  font: calc(14px + (26 - 14) * ((100vw - 300px) / (1600 - 300))) 'Titillium Web', sans-serif;
```
Comment (mithasgoogle) [inline portfolio/src/main/webapp/style.css:8]: LGTM -- Taylor's got great CSS intuition, will defer to her feedback here :)

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -62,8 +75,9 @@ function createCommentElement(comment) {
 }
 
 /** Tells the server to delete the comment. */
-function deleteComment(comment) {
+async function deleteComment(comment) {
```
Comment (mithasgoogle) [inline portfolio/src/main/webapp/script.js:78]: Hm, why does this need to be async?

File: portfolio/src/main/webapp/script.js [hunk 4]
```diff
@@ -33,12 +33,25 @@ function addRandomFact() {
...
-
+  
 /** Fetches comments from the server and adds it to the DOM. **/
-async function loadComments() {
-  const response = await fetch('/data');
+async function loadComments() { 
+  // Gets the maximum number of comments displayed.
+  let maxCommentsElement = document.getElementById('num-comments');
+  let maxComments = maxCommentsElement.value;
+  if (!maxComments) {
```
Comment (mithasgoogle) [inline portfolio/src/main/webapp/script.js:42]: Add a comment explaining in plain English what would trigger if vs. else block -- parsing/following the code can be hard when reviewer lacks context.

**Comment by mithasgoogle** [comment]
Will let Taylor merge pull request upon her feedback and approval.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 1 above)
Comment (sarahc7) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:93]: I wanted to return the default value for empty comments, but I think I should check that condition in `doPost`. I'll make that change.

File: portfolio/src/main/webapp/script.js (hunk 3 above)
Comment (sarahc7) [reply portfolio/src/main/webapp/script.js:78]: I need to wait for the comment to be deleted before I call `loadComments`.

File: portfolio/src/main/webapp/script.js (hunk 4 above)
Comment (sarahc7) [reply portfolio/src/main/webapp/script.js:42]: Done!

Comment (taylorwan) [reply portfolio/src/main/webapp/script.js:42]: Awesome use of local storage!

Optional: You can also write this without nesting if statements, which may be a little more readable. Something like this, if I'm not mis-reading your code:

```
if (maxCommentsElement.value) {
  localStorage.setItem('num-comments', maxCommentsElement.value);
} else if (localStorage.getItem('num-comments')) {
  maxCommentsElement.setAttribute('value', localStorage.getItem('num-comments'));
} else {
  localStorage.setItem('num-comments', 5);
}
```

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 5]
```diff
@@ -66,7 +71,7 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     long timestamp = System.currentTimeMillis();
 
     // Store the comment.
-    if (text != null) {
+    if (text != null && !text.equals("")) {
```
Comment (taylorwan) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:74]: There's a Java function called isEmpty() that I like, it makes the code very readable :)

Here's some examples of how it's used: https://beginnersbook.com/2017/10/java-string-isempty-method-with-example/#:~:text=Java%20String%20isEmpty()%20method%20checks%20whether%20a%20String%20is,of%20the%20string%20is%200.

There's also isBlank, but might be more than you need in this case: https://stackoverflow.com/questions/23419087/stringutils-isblank-vs-string-isempty

Comment (taylorwan) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:74]: There's a Java function called isEmpty() that I like, it makes the code very readable :)

Here's some examples of how it's used: https://beginnersbook.com/2017/10/java-string-isempty-method-with-example/

There's also `isBlank()`, which comes in handy if you don't want to save empty comments! https://stackoverflow.com/questions/23419087/stringutils-isblank-vs-string-isempty

File: portfolio/src/main/webapp/style.css (hunk 2 above)
Comment (taylorwan) [reply portfolio/src/main/webapp/style.css:8]: Responsive fonts are awesome - this is fine as long as it resizes well! Could you add a comment that explains the math?

File: portfolio/src/main/webapp/script.js [hunk 6]
```diff
@@ -33,12 +33,25 @@ function addRandomFact() {
...
+    if (!localStorage.getItem('num-comments'))
+      localStorage.setItem('num-comments', 5);
+    maxCommentsElement.setAttribute('value', localStorage.getItem('num-comments'));
+  } else {
+    localStorage.setItem('num-comments', maxComments);
+  }
```
Comment (taylorwan) [inline portfolio/src/main/webapp/script.js]: Oops, accidentally deleted my comment here. You could also make this not nested (and more readable) by doing something like this, if I'm reading your code correctly:

```
if (maxCommentsElement.value) {
  localStorage.setItem('num-comments', maxCommentsElement.value);
} else if (localStorage.getItem('num-comments')) {
  maxCommentsElement.setAttribute('value', localStorage.getItem('num-comments'));
} else {
  localStorage.setItem('num-comments', 5);
}

```

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 1 above)
Comment (taylorwan) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Gah! Deleting all my comments by accident - oops!

There's a Java function called `isEmpty()` that I like - it makes code very readable :) In this case, you could also use `isBlank()`, since it would prevent users from submitting empty comments.

isBlank: https://stackoverflow.com/questions/23419087/stringutils-isblank-vs-string-isempty
isEmpty: https://beginnersbook.com/2017/10/java-string-isempty-method-with-example/
